src/amqp-client.cr

```diff
@@ -29,8 +29,8 @@ class AMQP::Client
 
   alias TLSContext = OpenSSL::SSL::Context::Client | Bool | Nil
 
-  def self.start(url : String | URI, & : AMQP::Client::Connection -> _)
-    conn = self.new(url).connect
+  def self.start(url : String | URI, connection_information : ConnectionInformation? = nil, & : AMQP::Client::Connection -> _)
```

Comment:
constructors that takes an URI is expected to give options in the query params, like: `amqp://localhost?connection_name=helloworld&connection_product=foo&connection_product_version=1.2.3`

Reply:
Ok, so no need for connection_information here?

Reply:
No, but you should construct a `ConnectiontInformation` from the parameters in the `URI`, but in the constructor accepting an `URI`.